sawmill-core/src/test/java/io/logz/sawmill/processors/ExternalMappingProcessorRefreshTest.java

```diff
@@ -0,0 +1,102 @@
+package io.logz.sawmill.processors;
+
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.junit.WireMockClassRule;
+import io.logz.sawmill.Doc;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+import static io.logz.sawmill.utils.DocUtils.createDoc;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ExternalMappingProcessorRefreshTest {
+
+    public static final String SOURCE_FIELD_NAME = "author";
+    public static final String TARGET_FIELD_NAME = "books";
+
+    public static final String BOOKS_MAPPING = "/books";
+
+    private static Integer port;
+
+    @ClassRule
+    public static WireMockClassRule wireMockRule = new WireMockClassRule(WireMockConfiguration.wireMockConfig().dynamicPort());
+
+    @Rule
+    public WireMockClassRule instanceRule = wireMockRule;
+
+    @BeforeClass
+    public static void setUp() {
+        port = wireMockRule.port();
+    }
+
+    @Test
+    public void testMappingRefresh() throws MalformedURLException, InterruptedException {
```

Comment:
Nice test! 👏 
I am always afraid of those delicate timing tricks, like checking the first processing is empty while hoping 50 milliseconds did not pass yet. I wonder if we should change the way we test here:
1. Some unit tests on the refresh logic itself (directly invoking the method, no scheduled task) with different valid and invalid file inputs and make sure the map that was created is as expected
2. One test for coverage of the scheduled task - with is basically creating the processor with an empty map, and wait until it gets one refresh with a valid map (basically what you have here between lines 40-60, without the first assertion of `isEmpty()`)